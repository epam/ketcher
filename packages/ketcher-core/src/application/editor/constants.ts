export const canvasSelector = '#polymer-editor-canvas';
export const drawnStructuresSelector = '.drawn-structures';
