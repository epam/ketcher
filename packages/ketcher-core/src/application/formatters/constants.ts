export const macromoleculesFilesInputFormats = {
  ket: 'chemical/x-indigo-ket',
  mol: 'chemical/x-mdl-molfile',
  seq: {
    rna: 'chemical/x-rna-sequence',
    dna: 'chemical/x-dna-sequence',
    peptide: 'chemical/x-peptide-sequence',
  },
  fasta: {
    rna: 'chemical/x-rna-fasta',
    dna: 'chemical/x-dna-fasta',
    peptide: 'chemical/x-peptide-fasta',
  },
};
