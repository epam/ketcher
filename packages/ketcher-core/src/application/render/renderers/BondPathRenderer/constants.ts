import { Vec2 } from 'domain/entities';
import { HalfEdge } from 'application/render/view-model/HalfEdge';
import { BondType } from 'domain/entities/CoreBond';

export type BondSVGPath = {
  d: string;
  attrs: Record<string, string | number>;
};

export type BondVectors = {
  startPosition: Vec2;
  endPosition: Vec2;
  firstHalfEdge: HalfEdge;
  secondHalfEdge: HalfEdge;
};

export const BondWidth = 2;
export const StereoBondWidth = 6;
export const BondSpace = 6;
export const LinesOffset = BondSpace / 2;

export const BondDashArrayMap = {
  [BondType.Aromatic]: '6',
  [BondType.SingleDouble]: '6',
  [BondType.SingleAromatic]: '4 4 1 4',
  [BondType.DoubleAromatic]: '4 4 1 4',
  [BondType.Any]: '6',
  [BondType.Hydrogen]: '3',
};
