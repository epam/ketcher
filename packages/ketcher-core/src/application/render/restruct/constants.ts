export const VALENCE_MAP = {
  0: '0',
  1: 'I',
  2: 'II',
  3: 'III',
  4: 'IV',
  5: 'V',
  6: 'VI',
  7: 'VII',
  8: 'VIII',
  9: 'IX',
  10: 'X',
  11: 'XI',
  12: 'XII',
  13: 'XIII',
  14: 'XIV',
};
