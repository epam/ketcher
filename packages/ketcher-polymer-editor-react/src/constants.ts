/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

export const MONOMER_LIBRARY_FAVORITES = 'FAVORITES';

export const MONOMER_TYPES = {
  PEPTIDE: 'PEPTIDE',
  CHEM: 'CHEM',
  RNA: 'RNA',
} as const;
export type LibraryNameType =
  | typeof MONOMER_LIBRARY_FAVORITES
  | keyof typeof MONOMER_TYPES;

export const EditorClassName = 'Ketcher-polymer-editor-root';
export const EditorQuerySelector = `.${EditorClassName}`;

export const preview = {
  width: 230,
  height: 230,
  gap: 5,
} as const;

export enum MonomerGroups {
  SUGARS = 'Sugars',
  BASES = 'Bases',
  PHOSPHATES = 'Phosphates',
}

export enum MonomerGroupCodes {
  R = 'R',
  A = 'A',
  C = 'C',
  G = 'G',
  T = 'T',
  U = 'U',
  X = 'X',
  P = 'P',
}

export const MonomerCodeToGroup: Record<MonomerGroupCodes, MonomerGroups> = {
  R: MonomerGroups.SUGARS,
  A: MonomerGroups.BASES,
  C: MonomerGroups.BASES,
  G: MonomerGroups.BASES,
  T: MonomerGroups.BASES,
  U: MonomerGroups.BASES,
  X: MonomerGroups.BASES,
  P: MonomerGroups.PHOSPHATES,
} as const;
