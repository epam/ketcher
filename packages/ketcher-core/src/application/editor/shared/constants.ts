/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { IMAGE_KEY, MULTITAIL_ARROW_KEY } from 'domain/constants';

export const SgContexts = {
  Fragment: 'Fragment',
  Multifragment: 'Multifragment',
  Bond: 'Bond',
  Atom: 'Atom',
  Group: 'Group',
};

export const selectionKeys = [
  'atoms',
  'bonds',
  'frags',
  'sgroups',
  'rgroups',
  'rgroupAttachmentPoints',
  'rxnArrows',
  'rxnPluses',
  'simpleObjects',
  'texts',
  IMAGE_KEY,
  MULTITAIL_ARROW_KEY,
] as const;

export const defaultBondThickness = 2;

export enum MonomerGroups {
  SUGARS = 'Sugars',
  BASES = 'Bases',
  PHOSPHATES = 'Phosphates',
  PEPTIDES = 'Amino acids',
  NUCLEOTIDES = 'Nucleotides',
}

export enum MonomerGroupCodes {
  R = 'R',
  A = 'A',
  C = 'C',
  G = 'G',
  T = 'T',
  U = 'U',
  X = 'X',
  P = 'P',
}

export const MonomerCodeToGroup: Record<MonomerGroupCodes, MonomerGroups> = {
  R: MonomerGroups.SUGARS,
  A: MonomerGroups.BASES,
  C: MonomerGroups.BASES,
  G: MonomerGroups.BASES,
  T: MonomerGroups.BASES,
  U: MonomerGroups.BASES,
  X: MonomerGroups.BASES,
  P: MonomerGroups.PHOSPHATES,
} as const;
