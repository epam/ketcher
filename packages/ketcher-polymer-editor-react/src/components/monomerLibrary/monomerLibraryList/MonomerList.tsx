/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { useState } from 'react';
import { MonomerGroup } from '../monomerLibraryGroup';
import { useAppSelector } from 'hooks';
import { MonomerListContainer } from './styles';
import { IMonomerListProps } from './types';
import {
  selectFilteredMonomers,
  selectMonomerGroups,
  selectMonomersInCategory,
  selectMonomersInFavorites,
} from 'state/library';
import { MONOMER_LIBRARY_FAVORITES } from '../../../constants';
import { MonomerItemType } from '../monomerLibraryItem/types';

export type Group = {
  groupItems: Array<MonomerItemType>;
  groupTitle?: string;
};

const MonomerList = ({ onItemClick, libraryName }: IMonomerListProps) => {
  const monomers = useAppSelector(selectFilteredMonomers);
  const items =
    libraryName !== MONOMER_LIBRARY_FAVORITES
      ? selectMonomersInCategory(monomers, libraryName)
      : selectMonomersInFavorites(monomers);

  const groups = selectMonomerGroups(items);

  const [selectedMonomers, setSelectedMonomers] = useState('');

  const selectItem = (monomer: MonomerItemType) => {
    setSelectedMonomers(monomer.label);
  };

  return (
    <MonomerListContainer>
      {groups.map(({ groupItems, groupTitle }, index) => {
        if (groupTitle === '.') {
          return (
            <MonomerGroup
              key={index}
              items={groupItems}
              onItemClick={onItemClick || selectItem}
              selectedMonomerLabel={selectedMonomers}
            />
          );
        }

        return (
          <MonomerGroup
            key={groupTitle}
            title={groupTitle}
            items={groupItems}
            onItemClick={onItemClick || selectItem}
            selectedMonomerLabel={selectedMonomers}
          />
        );
      })}
    </MonomerListContainer>
  );
};

export { MonomerList };
