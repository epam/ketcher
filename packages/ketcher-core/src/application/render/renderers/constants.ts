import { KetMonomerClass } from 'application/formatters';

export const MONOMER_SYMBOLS_IDS = {
  [KetMonomerClass.AminoAcid]: {
    hover: '#peptide-hover',
    selected: '#peptide-selection',
    body: '#peptide',
  },
  [KetMonomerClass.CHEM]: {
    hover: '#chem-selection',
    selected: '#chem-selection',
    body: '#chem',
  },
  [KetMonomerClass.Sugar]: {
    hover: '#sugar-selection',
    selected: '#sugar-selection',
    body: '#sugar',
    variant: '#sugar-variant',
  },
  [KetMonomerClass.Base]: {
    hover: '#rna-base-selection',
    selected: '#rna-base-selection',
    body: '#rna-base',
    variant: '#rna-base-variant',
  },
  [KetMonomerClass.Phosphate]: {
    hover: '#phosphate-selection',
    selected: '#phosphate-selection',
    body: '#phosphate',
    variant: '#phosphate-variant',
  },
  [KetMonomerClass.RNA]: {
    hover: '#nucleotide-hover',
    selected: '#nucleotide-selection',
    body: '#nucleotide',
  },
};
