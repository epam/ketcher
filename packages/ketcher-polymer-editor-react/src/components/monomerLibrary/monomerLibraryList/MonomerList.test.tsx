/****************************************************************************
 * Copyright 2021 EPAM Systems
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 ***************************************************************************/

import { render, screen } from '@testing-library/react';
import { MONOMER_TYPES } from '../../../constants';

import { MonomerList } from './MonomerList';

describe('Monomer List', () => {
  const onItemClick = jest.fn();
  const initialState = {
    library: {
      searchFilter: '',
      favorites: {},
      monomers: [
        {
          props: {
            BranchMonomer: 'false',
            MonomerCaps: '[R1]H',
            MonomerCode: '',
            MonomerName: 'Phe4SD',
            MonomerNaturalAnalogCode: 'F',
            MonomerType: 'PEPTIDE',
            Name: '(2S)-2-amino-3-{4-[(4S)-2,6-dioxo-1,3-diazinane-4-amido]phenyl}propanoic acid',
          },
          struct: {},
        },
        {
          props: {
            BranchMonomer: 'false',
            MonomerCaps: '[R1]H',
            MonomerCode: '',
            MonomerName: 'PEG2',
            MonomerNaturalAnalogCode: '.',
            MonomerType: 'CHEM',
            Name: 'Diethylene Glycol',
          },
          struct: {},
        },
      ],
    },
  };

  it('should render correct with groups', () => {
    const view = render(
      withThemeAndStoreProvider(
        <MonomerList
          libraryName={MONOMER_TYPES.PEPTIDE}
          onItemClick={onItemClick}
        />,
        initialState,
      ),
    );

    const currentItem = screen.getByText('Phe4SD');
    expect(currentItem).toBeInTheDocument();

    expect(view).toMatchSnapshot();
  });

  it('should render correct without groups', () => {
    const view = render(
      withThemeAndStoreProvider(
        <MonomerList
          libraryName={MONOMER_TYPES.CHEM}
          onItemClick={onItemClick}
        />,
        initialState,
      ),
    );

    const currentItem = screen.getByText('PEG2');
    expect(currentItem).toBeInTheDocument();

    expect(view).toMatchSnapshot();
  });
});
