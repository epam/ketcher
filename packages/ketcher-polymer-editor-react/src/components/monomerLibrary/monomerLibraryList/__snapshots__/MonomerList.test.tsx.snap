// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Monomer List should render correct with groups 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-3whre8"
      >
        <div
          class="css-1jn33hb"
        >
          <div
            class="css-a0lsh8"
          >
            <span>
              F
            </span>
          </div>
          <div
            class="css-u8jo8f"
          >
            <div
              class="css-w9donl"
              data-testid="F"
            >
              <span>
                Phe4SD
              </span>
              <div
                class="star "
              >
                ★
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-3whre8"
    >
      <div
        class="css-1jn33hb"
      >
        <div
          class="css-a0lsh8"
        >
          <span>
            F
          </span>
        </div>
        <div
          class="css-u8jo8f"
        >
          <div
            class="css-w9donl"
            data-testid="F"
          >
            <span>
              Phe4SD
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Monomer List should render correct without groups 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-3whre8"
      >
        <div
          class="css-1jn33hb"
        >
          <div
            class="css-u8jo8f"
          >
            <div
              class="css-1gz3ldw"
              data-testid="."
            >
              <span>
                PEG2
              </span>
              <div
                class="star "
              >
                ★
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-3whre8"
    >
      <div
        class="css-1jn33hb"
      >
        <div
          class="css-u8jo8f"
        >
          <div
            class="css-1gz3ldw"
            data-testid="."
          >
            <span>
              PEG2
            </span>
            <div
              class="star "
            >
              ★
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
